classes/class-uagb-post-assets.php

```diff
@@ -226,22 +226,87 @@ public function __construct( $post_id ) {
 			$this->file_generation              = UAGB_Helper::$file_generation;
 			$this->is_allowed_assets_generation = $this->allow_assets_generation();
 		}
-
 		// Set other options.
 		$this->load_gfonts_locally = UAGB_Admin_Helper::get_admin_settings_option( 'uag_load_gfonts_locally', 'disabled' );
 		$this->preload_local_fonts = UAGB_Admin_Helper::get_admin_settings_option( 'uag_preload_local_fonts', 'disabled' );
 
 		if ( $this->is_allowed_assets_generation ) {
 			global $post;
 			$this_post = $this->preview ? $post : get_post( $this->post_id );
+			if ( function_exists( 'wp_is_block_theme' ) && wp_is_block_theme() ) {
+				$what_post_type = $this->determine_template_post_type();
+				$this->prepare_assets_for_templates_based_post_type( $what_post_type );
+			}
 			$this->prepare_assets( $this_post );
 			if ( $this->preview ) { // Load CSS only in preview mode of block editor.
 				$this->prepare_ast_custom_layout_post_assets();
 			}
 			$content = get_option( 'widget_block' );
 			$this->prepare_widget_area_assets( $content );
 		}
+	}
+
+	/**
+	 * Determine template post type function.
+	 *
+	 * @since x.x.x
+	 * @return string
+	 */
+	private function determine_template_post_type() {
+		$conditional_to_post_type = array(
+			'is_archive'    => 'archive',
+			'is_attachment' => 'attachment',
+			'is_author'     => 'author',
+			'is_category'   => 'category',
+			'is_date'       => 'date',
+			'is_embed'      => 'embed',
+			'is_front_page' => 'home',
+			'is_home'       => 'home',
+			'is_page'       => 'page',
+			'is_paged'      => 'paged',
+			'is_search'     => 'search',
+			'is_single'     => 'single',
+			'is_singular'   => 'singular',
+			'is_tag'        => 'tag',
+		);
 
+		$what_post_type = '404'; // Default to '404' if no condition matches.
+
+		foreach ( $conditional_to_post_type as $conditional => $post_type ) {
+			if ( $conditional() ) {
+				$what_post_type = $post_type;
+				break;
+			}
+		}
+
+		return $what_post_type;
+	}
+
+
+	/**
+	 * Generates assets for templates based on post type.
+	 *
+	 * @param string $post_type of current template.
+	 * @since x.x.x
+	 * @return void
+	 */
+	public function prepare_assets_for_templates_based_post_type( $post_type ) {
+		$template_slug = $post_type;
+		if ( is_singular() && is_page() ) { // will return true if you are using a static page is a homepage ( from settings > reading ).
+			// Run only if you are on the main website URL i.e example.com.
+			$template_slug = 'page';
+		} elseif ( ! ( is_front_page() && is_home() ) && is_home() ) {   // Blog page.
```

Comment:
Will this yield us the same result?

```suggestion
		} elseif ( is_home() && ! is_front_page() ) {   // Blog page.
```

Reply:
Resolved